modules/core/arrow-core-data/src/main/kotlin/arrow/typeclasses/Semialign.kt

```diff
@@ -0,0 +1,54 @@
+package arrow.typeclasses
+
+import arrow.Kind
+import arrow.core.Ior
+import arrow.core.identity
+
+/**
+ * A type class used for aligning of functors with non-uniform shapes.
+ *
+ * Note: Instances need to override either one of align/unlign here, otherwise a Stackoverflow exception will occur at runtime!
+ */
+interface Semialign<F> : Functor<F> {
+  /**
+   * Combines two structures by taking the union of their shapes and using Ior to hold the elements.
+   *
+   *  * {: data-executable='true'}
+   *
+   * ```kotlin:ank
+   * import arrow.core.extensions.*
+   * import arrow.core.extensions.listk.semialign.semialign
+   *
+   * fun main(args: Array<String>) {
+   *   //sampleStart
+   *   val result = ListK.semialign().run {
+   *    align(listOf("A", "B").k(), listOf(1, 2, 3).k())
+   *   }
+   *   //sampleEnd
+   *   println(result)
+   * }
+   * ```
+   */
+  fun <A, B> align(left: Kind<F, A>, right: Kind<F, B>): Kind<F, Ior<A, B>> = alignWith(::identity, left, right)
+
+  /**
+   * Combines two structures by taking the union of their shapes and combining the elements with the given function.
+   *
+   *    *  * {: data-executable='true'}
```

Comment:
Not sure how this would render, but can you remove the leading two "*"? Also in the above one there is one extra. I think the line gets removed anyway but this is kind of hard to test beforehand

Reply:
yes, thanks that was a copy paste error